Math/NumberTheory/GaussianIntegers.hs

```diff
@@ -0,0 +1,233 @@
+-- |
+-- Module:      Math.NumberTheory.GaussianIntegers
+-- Copyright:   (c) 2016 Chris Fredrickson
+-- Licence:     MIT
+-- Maintainer:  Chris Fredrickson <chris.p.fredrickson@gmail.com>
+-- Stability:   Provisional
+-- Portability: Non-portable (GHC extensions)
+--
+-- This module exports functions for manipulating Gaussian integers, including
+-- computing their prime factorisations.
+--
+
+{-# LANGUAGE BangPatterns #-}
+module Math.NumberTheory.GaussianIntegers (
+    GaussianInteger((:+)),
+    ι,
+    real,
+    imag,
+    conjugate,
+    norm,
+    divModG,
+    divG,
+    modG,
+    quotRemG,
+    quotG,
+    remG,
+    (.^),
+    isPrime,
+    primes,
+    gcdG,
+    gcdG',
+    findPrime,
+    findPrime',
+    factorise,
+) where
+
+import qualified Math.NumberTheory.Moduli as Moduli
+import qualified Math.NumberTheory.Powers as Powers
+import qualified Math.NumberTheory.Primes.Factorisation as Factorisation
+import qualified Math.NumberTheory.Primes.Sieve as Sieve
+import qualified Math.NumberTheory.Primes.Testing as Testing
+
+infix 6 :+
+infixr 8 .^
+-- |A Gaussian integer is a+bi, where a and b are both integers.
+data GaussianInteger = (:+) { real :: !Integer, imag :: !Integer } deriving (Eq)
+
+ι :: GaussianInteger
+ι = 0 :+ 1
+
+instance Show GaussianInteger where
+    show (a :+ b) = show a ++ op ++ b' ++ "ι"
+        where op = if b >= 0 then "+" else "-"
+              b' = if abs b /= 1 then show (abs b) else ""
```

Comment:
(Debatable.) IMHO it is worth to change `Show` instance to produce 
`[1+ι,3,1+2*ι,2+ι,7,11,2+3*ι,3+2*ι]` instead of `[1+ι,3+0ι,1+2ι,2+ι,7+0ι,11+0ι,2+3ι,3+2ι]`. What is your opinion?